### PR TITLE
Implement new UEI lookup flow

### PR DESCRIPTION
**Pull request by heymatthenry**
This PR implements the new flow that allows users to proceed creating an audit even if they don't have a valid UEI (GSA-TTS/FAC#285)

### Connection error state

I'd suggest trying it once not logged in to see the API failure state (it'll `401` but the connection failure status doesn't differentiate between all of the ways the API call could fail). 

image

If you click the `Go Back` button, it should just dismiss the modal and let you try again. If you click the other one it takes you to the no UEI flow:

image

### Invalid UEI state

To see the other states, you'll need to log in [at the homepage](https://federalist-9a617ff8-042d-4076-9581-bb999f9c6639.app.cloud.gov/preview/gsa-tts/fac-frontend/mh-no-uei-285/) first, but once you're logged in you can skip to the [Auditee Information](https://federalist-9a617ff8-042d-4076-9581-bb999f9c6639.app.cloud.gov/preview/gsa-tts/fac-frontend/mh-no-uei-285/audit/new/step-2/) page (which you probably _shouldn't_ be able to do but that's another ticket. 

If you put in a bad UEI, you get the UEI not recognized state

image

This one works opposite to the connection failure modal in that the primary button is the one to go ahead without a UEI and the secondary one closes the modal so the user can try again.

### Success

For the happy path, just put in a good UEI and you get:

image

If you continue, it hides the UEI & Auditee Name fields (and explanatory text) and replaces them with non-editable versions of the UEI & Name text:

image

### CODE REVIEW

File: src/js/check-ueid.js [hunk 1]
```diff
@@ -39,50 +37,180 @@ function handleAuditeeResponse(data) {
...
+        <p>You may also continue without the UEI, and you will be prompted to update the UEI before you can submit your audit.</p>
+      `,
+      buttons: {
+        primary: {
+          text: `Continue`,
+        },
+        secondary: { text: `Go back` },
+      },
+    },
+  };
```
Comment (heymatthenry) [inline src/js/check-ueid.js:184]: I'd like to have as much of this content as possible in the yaml so it's easier to edit, but in this instance, there just wasn't a good way to get that working with the modal. Populating the text in the modal via DOM manipulation was the least bad option.

File: .pa11yci [hunk 2]
```diff
@@ -12,6 +12,6 @@
       "htmlcs",
       "axe"
     ],
-    "hideElements": ".usa-select, .usa-step-indicator__segment-label"
+    "hideElements": ".usa-select, .usa-step-indicator__segment-label, .is-hidden"
```
Comment (heymatthenry) [inline .pa11yci:15]: `pa11y` was giving a false positive on the empty modal shell even though it's not in the accessibility tree.

**Comment by jasnakai** [comment]
The Connection Error State looks good.
The error message shown on continue needs more line spacing.
Screen Shot 2022-07-22 at 10 58 32 AM
I am unable to login via login.gov on the preview. This is the error message I get:
Screen Shot 2022-07-22 at 10 59 37 AM

**Review by jasnakai — APPROVED** [review]
Looks good!

**Comment by rachelschneiderman** [comment]
I am unable to login via login.gov on the preview. This is the error message I get: Screen Shot 2022-07-22 at 10 59 37 AM

@jasnakai Were you able to get logged in? I am seeing the same issue

**Comment by heymatthenry** [comment]
Thanks, @rachelschneiderman. @jasnakai brought this up as well. We're still working out how best to demo/preview/QA things on Federalist (additional context: GSA-TTS/FAC#370). I walked through the flow with @petermarks-gsa yesterday, so we're in good shape there. 

Even though this is merged, do feel free to add any thoughts about code stuff.

**Comment by rachelschneiderman** [comment]
Thanks, Matt! I was meaning to quote Jason's response in that first comment that I am also seeing that error locally and was just wondering what I should do to resolve that error?

**Pull request merged**